src/components/Products/Products.tsx

```diff
@@ -1,41 +1,40 @@
-import { collection, getDocs } from 'firebase/firestore';
-import { useEffect, useState } from 'react';
-import { db } from '../../firebase';
+import SearchBar from '../../layout/search-bar.layout'
+import ProductCardItem from './product-card/product-card-item'
+import { getProducts } from './products.service'
+import { ProductType } from '../../types/product.type'
+import { useEffect, useState } from 'react'
 
-export const Products = () => {
-    const [productName, setProductName] = useState('');
-    const [price, setPrice] = useState();
-    const [quantity, setQuantity] = useState();
-    const [description, setDescription] = useState('')
-    const [products, setProducts] = useState<any[]>([])
-    const productsCollectionRef = collection(db, 'products')
 
+export default function Products() {
+  const [productID, setProductID] = useState('')
+  const [productName, setProductName] = useState('')
+  const [price, setPrice] = useState()
+  const [quantity, setQuantity] = useState()
+  const [description, setDescription] = useState('')
+  const [products, setProducts] = useState<ProductType[]>([])
 
-    useEffect(() => {
-        const getProducts = async () => {
-            const data = await getDocs(productsCollectionRef)
+  useEffect(() => {
+    const productSubscription = getProducts().subscribe((products) => setProducts(products))
+    return () => productSubscription.unsubscribe()
+  }, [])
 
-            setProducts((data.docs.map((doc: any) => ({ ...doc.data(), id: doc.id }))))
+  return (
+    <>
+      <SearchBar />
+      <ul>
+        {products.map((product: ProductType) => {
+          return (
+            <li><ProductCardItem
+              productName={product.productName}
```

Comment:
product id and product color are mandatory fiels, so please provide an ID, if product color should not be mandatory set a default value if no color is passed

Reply:
done